docs/ko/docs/tutorial/testing.md

```diff
@@ -0,0 +1,243 @@
+
+///
+
+## 테스트 분리하기
+
+실제 애플리케이션에서는 테스트를 별도의 파일로 나누는 경우가 많습니다.
+
+
+그리고 **FastAPI** 애플리케이션도 여러 파일이나 모듈 등으로 구성될 수 있습니다.
+
+### **FastAPI** app 파일
+
+[Bigger Applications](bigger-applications.md){.internal-link target=_blank} 에 묘사된 파일 구조를 가지고 있는 것으로 가정해봅시다.
+
+```
+.
+├── app
+│   ├── __init__.py
+│   └── main.py
+```
+
+`main.py` 파일 안에 **FastAPI** app 을 만들었습니다:
+
+{* ../../docs_src/app_testing/main.py *}
+
+### 테스트 파일
+
+테스트를 위해 `test_main.py` 라는 파일을 생성할 수 있습니다. 이 파일은 동일한 Python 패키지(즉, `__init__.py` 파일이 있는 동일한 디렉터리)에 위치할 수 있습니다.
+
+``` hl_lines="5"
+.
+├── app
+│   ├── __init__.py
+│   ├── main.py
+│   └── test_main.py
+```
+
+파일들이 동일한 패키지에 위치해 있으므로, 상대 참조를 사용하여 `main` 에서 `app` 객체를 임포트 해올 수 있습니다.
+
+{* ../../docs_src/app_testing/test_main.py hl[3] *}
+
+
+...그리고 이전에 작성했던 것과 같은 테스트 코드를 작성할 수 있습니다.
+
+## 테스트: 확장된 예시
+
+이제 위의 예시를 확장하고 더 많은 세부 사항을 추가하여 다양한 부분을 어떻게 테스트하는지 살펴보겠습니다.
+
+### 확장된 FastAPI 애플리케이션 파일
+
+이전 예시와 동일한 파일 구조를 유지합니다:
+
+```
+.
+├── app
+│   ├── __init__.py
+│   ├── main.py
+│   └── test_main.py
+```
+
+이제 **FastAPI** 앱이 있는 `main.py` 파일에 몇 가지 다른 **경로 작업** 이 추가된 경우를 생각해봅시다.
+
+단일 오류를 반환할 수 있는 `GET` 작업이 있습니다.
+
+여러 다른 오류를 반환할 수 있는 `POST` 작업이 있습니다.
+
+두 *경로 동작* 모두 `X-Token` 헤더를 요구합니다.
+
+//// tab | Python 3.10+
+
+```Python
+{!> ../../docs_src/app_testing/app_b_an_py310/main.py!}
+```
+
+////
+
+//// tab | Python 3.9+
+
+```Python
+{!> ../../docs_src/app_testing/app_b_an_py39/main.py!}
+```
+
+////
+
+//// tab | Python 3.8+
+
+```Python
+{!> ../../docs_src/app_testing/app_b_an/main.py!}
+```
+
+////
+
+//// tab | Python 3.10+ non-Annotated
+
+/// tip | 팁
+
+될 수 있으면 `Annotated` 버전 사용을 권장합나다.
+
+///
+
+```Python
+{!> ../../docs_src/app_testing/app_b_py310/main.py!}
+```
+
+////
+
+//// tab | Python 3.8+ non-Annotated
+
+/// tip | 팁
+
+될 수 있으면 `Annotated` 버전 사용을 권장합나다.
+
+///
+
+```Python
+{!> ../../docs_src/app_testing/app_b/main.py!}
+```
+
+////
+
+### 확장된 테스트 파일
+
+이제는 `test_main.py` 를 확장된 테스트들로 수정할 수 있습니다:
+
+{* ../../docs_src/app_testing/app_b/test_main.py *}
+
+
+요청 설계가 `httpx` 의 설계를 기반으로 하고 있으므로, 클라이언트(TestClient)에 정보를 어떻게 전달해야 할지 모를 경우, 검색(Google)을 통해 `httpx` 에서 어떻게 처리하는지 찾아보면 알 수 있습니다.
+
+그 후, 테스트에서도 동일하게 적용하면 됩니다.
+
+예시:
+
+* *경로* 혹은 *쿼리* 매개변수를 전달하려면, URL 자체에 추가한다.
+* JSON body 를 전달하려면, 파이썬 객체 (예를들면 `dict`) 를 `json` 파라미터로 전달한다.
+* JSON 대신 *From Data* 를 보내야한다면, `data` 파라미터를 대신 전달한다.
+* *headers* 를 전달하려면, `headers` 파라미터에 `dict` 를 전달한다.
+* *cookies* 를 전달하려면, `cookies` 파라미터에 `dict` 를 전달한다.
+
+백엔드로 데이터를 어떻게 보내는지 정보를 더 얻으려면 (`httpx` 혹은 `TestClient` 를 이용해서) <a href="https://www.python-httpx.org" class="external-link" target="_blank">HTTPX documentation</a> 를 확인하세요.
+
+/// info | 정보
+
+`TestClient` 는 Pydantic 모델이 아니라 JSON 으로 변환될 수 있는 데이터를 받습니다.
+
+만약 테스트중 Pydantic 모델을 어플리케이션으로에 보내고 싶다면, [JSON Compatible Encoder](encoder.md){.internal-link target=_blank} 에 설명되어 있는 `jsonable_encoder` 를 사용할 수 있습니다.
```

Comment:
```suggestion
만약 테스트중 Pydantic 모델을 어플리케이션으로에 보내고 싶다면, [JSON 호환 가능 인코더](encoder.md){.internal-link target=_blank} 에 설명되어 있는 `jsonable_encoder` 를 사용할 수 있습니다.
```

It's [translated](https://fastapi.tiangolo.com/ko/tutorial/encoder/) so reflect it, please.